tests/ident/test_engine_parser_comet_2020_01_04.py

```diff
@@ -0,0 +1,145 @@
+    rt_lookup_path = Path(__file__).parent.parent / "data" / "BSA1_ursgal_lookup.csv.bz2"
+    db_path = (
+        Path(__file__).parent.parent / "data" / "test_Creinhardtii_target_decoy.fasta"
+    )
+
+    parser = Comet_2020_01_4_Parser(
+        input_file,
+        params={
+            "cpus": 2,
+            "rt_pickle_name": rt_lookup_path,
+            "database": db_path,
+            "modifications": [
+                {
+                    "aa": "M",
+                    "type": "opt",
+                    "position": "any",
+                    "name": "Oxidation",
+                },
+                {
+                    "aa": "C",
+                    "type": "fix",
+                    "position": "any",
+                    "name": "Carbamidomethyl",
+                },
+                {
+                    "aa": "*",
+                    "type": "opt",
+                    "position": "Prot-N-term",
+                    "name": "Acetyl",
+                },
+            ],
+            "omssa_mod_dir": Path(__file__).parent.parent / "data",
+        },
+    )
+
+
+def test_engine_parsers_comet_check_parser_compatibility():
+    msgf_parser_class = Comet_2020_01_4_Parser
+    input_file = Path(__file__).parent.parent / "data" / "BSA1_comet_2020_01_4.mzid"
+    assert msgf_parser_class.check_parser_compatibility(input_file) is True
+
+
+def test_engine_parsers_comet_check_parser_compatibility_fail_with_omssa_file():
+    msgf_parser_class = Comet_2020_01_4_Parser
+    input_file = (
+        Path(__file__).parent.parent
+        / "data"
+        / "test_Creinhardtii_QE_pH11_omssa_2_1_9.csv"
+    )
+    assert msgf_parser_class.check_parser_compatibility(input_file) is False
+
+
+def test_engine_parsers_comet_unify():
+    input_file = Path(__file__).parent.parent / "data" / "BSA1_comet_2020_01_4.mzid"
+    rt_lookup_path = Path(__file__).parent.parent / "data" / "BSA1_ursgal_lookup.csv.bz2"
+    db_path = Path(__file__).parent.parent / "data" / "BSA.fasta"
+
+    parser = Comet_2020_01_4_Parser(
+        input_file,
+        params={
+            "cpus": 2,
+            "rt_pickle_name": rt_lookup_path,
+            "database": db_path,
+            "modifications": [
+                {
+                    "aa": "M",
+                    "type": "opt",
+                    "position": "any",
+                    "name": "Oxidation",
+                },
+                {
+                    "aa": "C",
+                    "type": "fix",
+                    "position": "any",
+                    "name": "Carbamidomethyl",
+                },
+                {
+                    "aa": "*",
+                    "type": "opt",
+                    "position": "Prot-N-term",
+                    "name": "Acetyl",
+                },
+            ],
+            "omssa_mod_dir": Path(__file__).parent.parent / "data",
+        },
+    )
+    df = parser.unify()
+
+    assert len(df) == 60
```

Comment:
maybe add some identity assertion as well ? Number of Acetyl or something like that ...